R/score.R

```diff
@@ -110,53 +97,114 @@
 #' PLoS Comput Biol 15(2): e1006785. \doi{10.1371/journal.pcbi.1006785}
 #' @export
 
-score <- function(data,
-                  metrics = NULL,
-                  ...) {
+score <- function(data, ...) {
+  UseMethod("score")
+}
+
+#' @rdname score
+#' @export
+score.default <- function(data, ...) {
+  data <- validate(data)
+  score(data, ...)
+}
 
-  # preparations ---------------------------------------------------------------
-  if (is_scoringutils_check(data)) {
-    check_data <- data
-  } else {
-    check_data <- check_forecasts(data)
-  }
+#' @rdname score
+#' @export
+score.scoringutils_binary <- function(data, metrics = metrics_binary, ...) {
+  data <- validate(data)
+  data <- remove_na_observed_predicted(data)
+  forecast_unit <- attr(data, "forecast_unit")
+  metrics <- validate_metrics(metrics)
 
-  data <- check_data$cleaned_data
-  prediction_type <- check_data$prediction_type
-  forecast_unit <- check_data$forecast_unit
-  target_type <- check_data$target_type
+  # Extract the arguments passed in ...
+  args <- list(...)
+  lapply(seq_along(metrics), function(i, ...) {
```

Comment:
This looks like code that will be shared across every method?

Reply:
Yes, but with slight variations - so e.g. for quantile you'd have an additional quantile argument, for binary and point you don't need to do anything by forecast_unit as there is a 1:1 correspondence between forecast and score. 
I couldn't think of a clever way to make it one function that works for all methods...